ldms/src/ldmsd/ldmsd_stream.c

```diff
@@ -90,6 +90,8 @@ void ldmsd_stream_deliver(const char *stream_name, ldmsd_stream_type_t stream_ty
 			&& entity == NULL	/* data hasn't been parsed yet */
 			&& parser == NULL)	/* we haven't tried and failed already */
 		{
+			if (parser)
```

Comment:
parser has to be NULL to enter this path. This is just dead code I think.

Reply:
2nd+ trips through the loop parser is not null.

Reply:
but, yes, i'll drop this non fix and figure out how else we can get to losing that pointer.